internal/internal_test.go

```diff
@@ -545,6 +546,41 @@ func TestConfigValid(t *testing.T) {
 	}
 }
 
+func TestConfigValidWithPath(t *testing.T) {
+
+	m, err := plugins.New([]byte{}, "test", inmem.New())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tests := map[string]struct {
+		path string
+		want string
+	}{
+		"empty_path":               {path: "", want: "data.istio.authz.allow"},
+		"path_no_lt_slash":         {path: "test/allow/main", want: "data.test.allow.main"},
+		"path_with_leading_slash":  {path: "/test/allow/main", want: "data.test.allow.main"},
+		"path_with_trailing_slash": {path: "test/allow/main/", want: "data.test.allow.main"},
+		"path_with_lt_slash":       {path: "/test/allow/main/", want: "data.test.allow.main"},
+		"path_with_periods":        {path: "test/com.foo.istio.ingress/allow/main/", want: "data.test.com.foo.istio.ingress.allow.main"},
```

Comment:
I'm not sure that this is correct, shouldn't it be something more like `data.test["com.foo.istio.ingress"].main` (well, a ref with like `["data", "test", "com.foo.istio.ingress", "main]`, we don't need that intermediate string representation)

I think we need to do something more like what the server does over here: https://github.com/open-policy-agent/opa/blob/master/server/server.go#L2179-L2205 (maybe don't need to do the url escape stuff for this one) rather than the swap of all `/` to `.` and parsing the resulting string. It should split on `/` and build the ref from the segments.